packages/yourloops/pages/hcp/team-members-table.tsx

```diff
@@ -89,72 +95,108 @@ function MembersTableBody(props: TeamMembersProps): JSX.Element {
-        <div style={{ display: "flex", alignItems: "center", justifyContent: "center", width: "42px", height: "42px" }}>
-          <CircularProgress disableShrink size={17} />
-        </div>
-      );
+    let removeMemberButton: JSX.Element | null = null;
+    let rowClassName = "";
+    let icon: JSX.Element | null = null;
+
+    if (member.status === TeamMemberStatus.accepted) {
+      // Determine if the current user can change the admin role for this member
+      // - Must be an admin
+      // - Mustn't be the only admin for it's own entry
+      // - An update mustn't be in progress
+      const checkboxAdminDisabled = !userIsAdmin || (userIsTheOnlyAdministrator && userId === currentUserId) || updatingUser.length > 0;
+      if (updatingUser === userId) {
+        // Disabled while update in progress (backend api call in progress)
+        checkboxElement = (
+          <div style={{ display: "flex", alignItems: "center", justifyContent: "center", width: "42px", height: "42px" }}>
+            <CircularProgress disableShrink size={17} />
+          </div>
+        );
+      } else {
+        const handleSwitchRole = async (event: React.ChangeEvent<HTMLInputElement>): Promise<void> => {
+          const userId = event.target.name;
+          const isAdmin = event.target.checked;
+          setUpdatingUser(userId);
+          await onSwitchAdminRole(member, isAdmin ? TeamMemberRole.admin : TeamMemberRole.viewer);
+          setUpdatingUser("");
+        };
+        checkboxElement = (
+          <Checkbox
+            disabled={checkboxAdminDisabled}
+            id={`team-members-list-${team.id}-row-${userId}-role-checkbox`}
+            color="primary"
+            name={userId}
+            checked={isAdmin}
+            onChange={handleSwitchRole} />
+        );
+      }
     } else {
-      const handleSwitchRole = async (event: React.ChangeEvent<HTMLInputElement>): Promise<void> => {
-        const userId = event.target.name;
-        const isAdmin = event.target.checked;
-        setUpdatingUser(userId);
-        await onSwitchAdminRole(member, isAdmin ? TeamMemberRole.admin : TeamMemberRole.viewer);
-        setUpdatingUser("");
-      };
-      checkboxElement = (
-        <Checkbox
-          disabled={checkboxAdminDisabled}
-          id={`team-members-list-${team.id}-row-${userId}-role-checkbox`}
-          color="primary"
-          name={userId}
-          checked={isAdmin}
-          onChange={handleSwitchRole} />
+      rowClassName = props.classes?.tableRowPending ?? "";
+      icon = (
+        <Tooltip title={t("team-member-pending") as string} aria-label={t("team-member-pending")} placement="bottom">
+          <AccessTimeIcon />
+        </Tooltip>
       );
     }
 
-    let removeMemberButton: JSX.Element | null = null;
     if (userIsAdmin && userId !== currentUserId) {
       const handleClickRemoveMember = async (): Promise<void> => {
         await onShowRemoveTeamMemberDialog(member);
```

Comment:
I would expect that we manage differently the remove of effective members and the remove of pending members.

Reply:
No, for now, nothing is different I think.
@MarineDBL An advice on this one ?

Reply:
That's correct. Something that can be managed later on. 
Anyway we have to keep in mind that api calls will be different as we will have to disable the pending notification. I don't think there is a route defined for that in hydrophone.